### PR TITLE
PMM-6696 MongoDB clusters

### PR DESCRIPTION
**Pull request by tiagomotasantos**
[PMM-6696](https://jira.percona.com/browse/PMM-6696)

This PR adds MongoDB Clusters (PSMDB) to DBaaS.

Turned off `class-methods-use-this` rule in eslint, because it's the second PR I came across the need of removing this rule. I think it makes sense because we might need to use classes without instance properties/methods so we don't need to use `this`. For example, in order to protect the code against variations (more types of clusters can be added), I used a factory to create the appropriate service regarding the type of database the user chooses. The code only knows the DBClusterService abstraction and doesn't know the actual implementations. To achieve this I needed to use class methods to leverage inheritance/polymorphism. So in the future when we have to add PostgreSQL clusters we just need to add a new service and the rest of the code doesn't need to change. No need for conditions.
If you don't agree with removing this rule I can add exceptions to some files.

Plus this PR renames everything with `XtraDB` to `DBCluster`. The only thing that remains with XtraDB name is the service since is the only point of variation as mentioned above. If you want to see just the MongoDB addition without the name changes take a look at [this commit](https://github.com/percona/grafana-dashboards/pull/1026/commits/e35d439df8f2b946bec95506854b4a38e03eb973).

https://github.com/Percona-Lab/pmm-submodules/pull/1214

### CODE REVIEW

**Review by lunaticusgreen — DISMISSED** [review]
Looks good enough, lets proceed it to the QA

File: pmm-app/src/pmm-dbaas/components/DBCluster/DBClusterService.factory.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+export class DBClusterServiceFactory {
+  static newDBClusterService(type: Databases): DBClusterService {
+    switch (type) {
+      case Databases.mysql:
+        return new XtraDBService();
+      case Databases.mongodb:
+        return new PSMDBService();
+      default:
+        return new XtraDBService();
```
[the remark this answers is not in the document]
Comment (tiagomotasantos) [reply pmm-app/src/pmm-dbaas/components/DBCluster/DBClusterService.factory.ts]: Done!

File: pmm-app/src/pmm-dbaas/components/DBCluster/AddDBClusterModal/DBClusterAdvancedOptions/DBClusterAdvancedOptions.test.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import { DBClusterResources, DBClusterTopology } from './DBClusterAdvancedOptions.types';
+
+describe('DBClusterAdvancedOptions::', () => {
+  it('renders correctly', () => {
+    const root = mount(<Form
+      onSubmit={jest.fn()}
+      render={(renderProps) => <DBClusterAdvancedOptions {...renderProps} />}
+    />);
+
+    expect(root.find('[data-qa="dbcluster-topology-field"]')).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (tiagomotasantos) [reply pmm-app/src/pmm-dbaas/components/DBCluster/AddDBClusterModal/DBClusterAdvancedOptions/DBClusterAdvancedOptions.test.tsx:16]: Thanks for reminding, need to update platform-core on pmm and will start using this!

**Review by lunaticusgreen — APPROVED** [review]
👍

**Pull request merged**